Sources/Bob/Commands/Bump/BumpCommand.swift

```diff
@@ -0,0 +1,117 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Bob.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import Vapor
+
+// Command used to bump up build numbers directly on GitHub
+public class BumpCommand {
+    
+    struct Constants {
+        static let branchSpecifier: String = "-b"
+    }
+    
+    fileprivate let gitHub: GitHub
+    fileprivate let defaultBranch: BranchName
+    fileprivate let plistPaths: [String]
+    fileprivate let message: String
+    fileprivate let author: Author
+    /// Initializer
+    ///
+    /// - Parameters:
+    ///   - config: Configuration to use to connect to github
+    ///   - defaultBranch: Default branch
+    ///   - plistPaths: Paths to .plist files to update. Path relative from the root of the repository
+    ///   - author: Commit author. Shows up in GitHub
+    ///   - messageFormat: Format for the commit message. `<version>` will be replaced with the version string
+    public init(gitHub: GitHub, defaultBranch: BranchName, plistPaths: [String], author: Author, message: String = "[General] Bumps version to <version> (<buildNumber>).") {
+        self.gitHub = gitHub
+        self.defaultBranch = defaultBranch
+        self.plistPaths = plistPaths
+        self.message = message
+        self.author = author
+    }
+}
+
+extension BumpCommand: Command {
+    
+    public var name: String {
+        return "bump"
+    }
+    
+    public var usage: String {
+        return "Bump up build number by typing `bump`. Specify a branch by typing `\(Constants.branchSpecifier) {branch}`, defaults to `" + self.defaultBranch.name + "`."
+    }
+    
+    public func execute(with parameters: [String], replyingTo sender: MessageSender) throws {
+        var params = parameters
+        
+        var branch: BranchName = self.defaultBranch
+        if let branchSpecifierIndex = params.index(where: { $0 == Constants.branchSpecifier }) {
+            guard params.count > branchSpecifierIndex + 1 else { throw "Branch name not specified after `\(Constants.branchSpecifier)`" }
+            branch = BranchName(params[branchSpecifierIndex + 1])
+            params.remove(at: branchSpecifierIndex + 1)
+            params.remove(at: branchSpecifierIndex)
+        }
+        
+        sender.send("One sec...")
+        let versionFiles = try self.gitHub.currentState(on: branch).items.filter { $0.path == plistPaths[0] }
+        
+        guard let versionAndBuildNumber = try versionFiles.first.map ({ item -> (String, String) in
+            
+            let content = try self.gitHub.content(forBlobWith: item.sha)
+            
+            let matcher = RegexMatcher(text: content)
+            var output = ("","")
+            
+            let versions = matcher.matches(stringMatching: PListHelpers.versionRegexString)
+            if let version = versions.first {
+                let versionString = PListHelpers.extractStringValue(from: version)
```

Comment:
If the `version` isn't found, the result will be `""`, and the command will change the version to an empty string. I believe we should `throw` here.